Marlin/src/HAL/HAL_STM32F1/HAL_spi_STM32F1.cpp

```diff
@@ -79,15 +79,25 @@ void spiBegin() {
  * @details
  */
 void spiInit(uint8_t spiRate) {
+  /**
+   * STM32F1 APB1 = 72MHz, APB2 = 36MHz, max SPI speed of this MCU if 18Mhz 
+   * STM32F1 has 3 SPI ports, SPI1 in APB1, SPI2/SPI3 in APB2
+   * so the minimum prescale of SPI1 is DIV4, SPI2/SPI3 is DIV2
+  */
+  #if SPI_DEVICE == 1
+    #define SPI_CLOCK_MAX SPI_CLOCK_DIV4
+  #else
+    #define SPI_CLOCK_MAX SPI_CLOCK_DIV2
+  #endif
```

Comment:
Hmm, you are limiting SPI speed but some devices may handle full speed (like eeproms, some sd cards too)

Which device doesnt handle the 72MHz ?

Reply:
https://www.stm32duino.com/viewtopic.php?t=1495 reports 36Mhz working for SPI1